components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/graphQL/GraphQLAPIHandler.java

```diff
@@ -0,0 +1,296 @@
+
+import java.io.IOException;
+import java.util.Base64;
+import java.net.URLDecoder;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.xml.stream.XMLStreamException;
+
+public class GraphQLAPIHandler extends AbstractHandler {
+
+    private static final String QUERY_PATH_STRING = "/?query=";
+    private static final String REST_SUB_REQUEST_PATH = "REST_SUB_REQUEST_PATH";
+    private static final String API_TYPE = "API_TYPE";
+    private static final String GRAPHQL_API = "GRAPHQL";
+    private static final String UNICODE_TRANSFORMATION_FORMAT = "UTF-8";
+    private static final String INVALID_QUERY = "INVALID QUERY";
+    private static final String SCOPE_ROLE_MAPPING = "ScopeRoleMapping";
+    private static final String SCOPE_OPERATION_MAPPING = "ScopeOperationMapping";
+    private static final String GRAPHQL_IDENTIFIER = "_graphQL";
+    private static final String CLASS_NAME_AND_METHOD = "_GraphQLAPIHandler_handleRequest";
+    private static final Log log = LogFactory.getLog(GraphQLAPIHandler.class);
+    private static GraphQLSchema schema = null;
+    private static Validator validator;
+    private String apiUUID;
+
+    public GraphQLAPIHandler() {
+
+        validator = new Validator();
+    }
+
+    public String getApiUUID() {
+
+        return apiUUID;
+    }
+
+    public void setApiUUID(String apiUUID) {
+
+        this.apiUUID = apiUUID;
+    }
+
+    public boolean handleRequest(MessageContext messageContext) {
+        try {
+            String payload;
+            String operationList = "";
+            Parser parser = new Parser();
+
+            ArrayList<String> operationArray = new ArrayList<>();
+
+            org.apache.axis2.context.MessageContext axis2MC = ((Axis2MessageContext) messageContext).
+                    getAxis2MessageContext();
+            String requestPath = ((Axis2MessageContext) messageContext).getProperties().
+                    get(REST_SUB_REQUEST_PATH).toString();
+            if (!requestPath.isEmpty()) {
+                String[] queryParams = ((Axis2MessageContext) messageContext).getProperties().
+                        get(REST_SUB_REQUEST_PATH).toString().split(QUERY_PATH_STRING);
+                if (queryParams.length > 1) {
+                    String queryURLValue = queryParams[1];
+                    payload = URLDecoder.decode(queryURLValue, UNICODE_TRANSFORMATION_FORMAT);
+                } else {
+                    RelayUtils.buildMessage(axis2MC);
+                    OMElement body = axis2MC.getEnvelope().getBody().getFirstElement();
+                    if (body != null && body.getFirstElement() != null) {
+                        payload = body.getFirstElement().getText();
+                    } else {
+                        log.debug("Invalid query parameter " + queryParams[0]);
```

Comment:
check for isDebugEnabled since you do a concatenation

Reply:
Fixed with 2a6fbdd3ac9ce54ca7bc94a45328a41826836a80